src/main/java/pl/piekoszek/gorskimatches/challange/ChallengeController.java

```diff
@@ -0,0 +1,99 @@
+package pl.piekoszek.gorskimatches.challange;
+
+import org.springframework.web.bind.annotation.*;
+import pl.piekoszek.gorskimatches.equation.EquationRandomizer;
+import pl.piekoszek.gorskimatches.token.EmailService;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("api/challenge")
+public class ChallengeController {
```

Comment:
There is a lot of `logic` in this Controller class, it should only handle HTTP related stuff and `Service` class should do rest.

You'll see why it's important, when we write more tests.